packages/channel-utils/src/date-utils.ts

```diff
@@ -1,24 +1,53 @@
 import { addInterval, getDayPriorTillTomorrow, getLastXMonths, getTomorrowStartOfDay } from '@repo/utils';
-import { prisma } from '@repo/database';
+import { type Insight, prisma, Tier } from '@repo/database';
+import { tierConstraints } from '@repo/mappings';
+
+interface LatestInsightResult {
+  date: Date;
+  adAccount?: {
+    organizations: {
+      tier: Tier;
+    }[];
+  };
+}
 
 export const timeRanges = async (initial: boolean, adAccountId: string): Promise<{ since: Date; until: Date }[]> => {
-  if (initial) {
-    const range = getLastXMonths();
-    return splitTimeRange(range.since, range.until);
-  }
   const latestInsight = await prisma.insight.findFirst({
-    select: { date: true },
+    select: { date: true, adAccount: { select: { organizations: { select: { tier: true } } } } },
     where: { adAccountId },
     orderBy: { date: 'desc' },
   });
-  const furthestDate = addInterval(new Date(), 'day', -maxTimePeriodDays);
-  const range = latestInsight ? getDayPriorTillTomorrow(latestInsight.date) : getDayPriorTillTomorrow(furthestDate);
-  return splitTimeRange(range.since, range.until);
+
+  const organizations = latestInsight?.adAccount?.organizations ?? [];
+
+  const organizationWithHighestTier =
+    organizations.length > 0
+      ? organizations.sort((a, b) => tierConstraints[b.tier].order - tierConstraints[a.tier].order)[0].tier
+      : undefined;
+
+  const maxRecency = tierConstraints[organizationWithHighestTier ?? Tier.Launch].maxRecency;
+
+  if (initial) {
+    const range = getLastXMonths();
+
+    if (!organizationWithHighestTier) {
+      return splitTimeRange(tierConstraints[Tier.Launch].maxRecency, range.since, range.until);
+    }
+
+    return splitTimeRange(maxRecency, range.since, range.until);
+  }
+  const furthestDate = addInterval(new Date(), 'day', -maxRecency);
+  const insightDateWithinLimit = latestInsight && latestInsight.date > furthestDate ? latestInsight.date : furthestDate;
+  const range = getDayPriorTillTomorrow(insightDateWithinLimit);
+  return splitTimeRange(maxRecency, range.since, range.until);
```

Comment:
```suggestion
  return testTimeRange(initial, latestInsight);
```
I am a bit confused. You have created a `testTimeRange` (probably a name that we need to change) function that you test, but the product code does not use it.